internal/service/projectapikey/resource_project_api_key.go

```diff
@@ -246,223 +205,112 @@ func resourceDelete(ctx context.Context, d *schema.ResourceData, meta any) diag.
-	}
-
-	projectAssignments, err := getAPIProjectAssignments(ctx, connV2, apiKeyOrgList, apiKeyID)
-	if err != nil {
-		return diag.FromErr(fmt.Errorf("error getting api key information: %s", err))
-	}
-
-	for _, apiKey := range projectAssignments {
-		_, _, err = connV2.ProgrammaticAPIKeysApi.RemoveProjectApiKey(ctx, apiKey.ProjectID, apiKeyID).Execute()
-		if err != nil {
-			return diag.FromErr(fmt.Errorf("error deleting project api key: %s", err))
-		}
-	}
-
-	if orgID != "" {
+	if details != nil && orgID != "" {
 		if _, _, err = connV2.ProgrammaticAPIKeysApi.DeleteApiKey(ctx, orgID, apiKeyID).Execute(); err != nil {
-			return diag.FromErr(fmt.Errorf("error unable to delete Key (%s): %s", apiKeyID, err))
+			return diag.FromErr(fmt.Errorf("error deleting project key (%s): %s", apiKeyID, err))
 		}
 	}
-
 	d.SetId("")
 	return nil
 }
 
 func resourceImportState(ctx context.Context, d *schema.ResourceData, meta any) ([]*schema.ResourceData, error) {
-	connV2 := meta.(*config.MongoDBClient).AtlasV2
-
-	parts := strings.SplitN(d.Id(), "-", 2)
-	if len(parts) != 2 {
-		return nil, errors.New("import format error: to import a api key use the format {project_id}-{api_key_id}")
-	}
-
-	projectID := parts[0]
-	apiKeyID := parts[1]
-
-	projectAPIKeys, _, err := connV2.ProgrammaticAPIKeysApi.ListProjectApiKeys(ctx, projectID).Execute()
-	if err != nil {
-		return nil, fmt.Errorf("couldn't import api key %s in project %s, error: %s", projectID, apiKeyID, err)
-	}
-	for _, val := range projectAPIKeys.GetResults() {
-		if val.GetId() == apiKeyID {
-			if err := d.Set("description", val.GetDesc()); err != nil {
-				return nil, fmt.Errorf("error setting `description`: %s", err)
-			}
-
-			if err := d.Set("public_key", val.GetPublicKey()); err != nil {
-				return nil, fmt.Errorf("error setting `public_key`: %s", err)
-			}
-
-			if projectAssignments, err := newProjectAssignment(ctx, connV2, apiKeyID); err == nil {
-				if err := d.Set("project_assignment", projectAssignments); err != nil {
-					return nil, fmt.Errorf("error setting  `project_assignment`: %s", err)
-				}
-			}
-
-			d.SetId(conversion.EncodeStateID(map[string]string{
-				"api_key_id": val.GetId(),
-			}))
-		}
-	}
+	d.SetId(conversion.EncodeStateID(map[string]string{
+		"api_key_id": d.Id(),
+	}))
 	return []*schema.ResourceData{d}, nil
 }
 
-func getFirstProjectIDFromAssignments(d *schema.ResourceData) (*string, error) {
-	if projectAssignments, ok := d.GetOk("project_assignment"); ok {
-		projectAssignmentList := ExpandProjectAssignmentSet(projectAssignments.(*schema.Set))
-		if len(projectAssignmentList) < 1 {
-			return nil, errors.New(errorNoProjectAssignmentDefined)
-		}
-		return admin.PtrString(projectAssignmentList[0].ProjectID), nil // can safely assume at least one assigment is defined because of schema definition
+func expandProjectAssignments(projectAssignments *schema.Set) map[string][]string {
```

Comment:
nit: move all the helper methods like expand/flatten to `model_project_api_key` ?

Reply:
model file created here: https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2737/commits/055eca62452330d081854e6704c9fb3925607e96